packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts

```diff
@@ -52,15 +54,16 @@ export async function setSourceApiVersion(componentSet: ComponentSet): Promise<v
 
   // Next, if it still is not set, set it to the highest API version supported by the target org.
   if (!sourceApiVersion) {
-    const orgApiVersion = await getOrgApiVersion();
-    sourceApiVersion = orgApiVersion ?? componentSet.sourceApiVersion;
+    const orgApiVersion = await exports.getOrgApiVersion();
+    sourceApiVersion = orgApiVersion;
```

Comment:
Is there a possibility that componentSet already has sourceApiVersion set when passed through this function?

Reply:
I'm not sure I understand the question.  When `setSourceApiVersion(componentSet: ComponentSet)` is called...
...in the scenario that the sourceApiVersion has been set via a metadata file, by the time `setSourceApiVersion()` is called, `componentSet.sourceApiVersion` has been set, and the function exits at line 43.

...in the scenario that the sourceApiVersion is not set, it wasn't set via a metadata file, and then we try to get it from sfdx-project.json, etc...

Let's discuss in our standup/huddle this morning.